dotnet/src/Connectors/Connectors.Memory.SqlServer.Classic/Core/SqlServerClient.cs

```diff
@@ -0,0 +1,486 @@
+    /// <param name="configuration">The configuration to use for the SQL Server database.</param>
+    public SqlServerClient(SqlConnection connection, SqlServerConfig configuration)
+    {
+        this._configuration = configuration;
+
+        this._connection = connection;
+    }
+
+    /// <inheritdoc />
+    public async Task CreateCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        collectionName = NormalizeIndexName(collectionName);
+
+        if (await this.DoesCollectionExistsAsync(collectionName, cancellationToken).ConfigureAwait(false))
+        {
+            // Collection already exists
+            return;
+        }
+
+        using (await this.OpenConnectionAsync(cancellationToken).ConfigureAwait(false))
+        {
+            using var command = this._connection.CreateCommand();
+
+            command.CommandText = $@"
+                    IF OBJECT_ID(N'{this.GetFullTableName($"{this._configuration.MemoryTableName}_{collectionName}")}', N'U') IS NULL
+                    CREATE TABLE {this.GetFullTableName($"{this._configuration.MemoryTableName}_{collectionName}")}
+                    (   [id] UNIQUEIDENTIFIER NOT NULL,
+                        [key] NVARCHAR(256)  NOT NULL,
+                        [metadata] TEXT,
+                        [embedding] TEXT,
+                        [timestamp] DATETIMEOFFSET,
+                        PRIMARY KEY ([id])
+                    );
+
+                    IF OBJECT_ID(N'{this.GetFullTableName($"{this._configuration.EmbeddingsTableName}_{collectionName}")}', N'U') IS NULL
+                    CREATE TABLE {this.GetFullTableName($"{this._configuration.EmbeddingsTableName}_{collectionName}")}
+                    (
+                        [memory_id] UNIQUEIDENTIFIER NOT NULL,
+                        [vector_value_id] [int] NOT NULL,
+                        [vector_value] [float] NOT NULL
+                    );
+
+                    IF OBJECT_ID(N'{NormalizeSQLObjectName(this._configuration.Schema)}.IXC_{$"{NormalizeSQLObjectName(this._configuration.EmbeddingsTableName)}_{collectionName}"}', N'U') IS NULL
+                    CREATE CLUSTERED COLUMNSTORE INDEX [IXC_{$"{NormalizeSQLObjectName(this._configuration.EmbeddingsTableName)}_{collectionName}]"}
+                    ON {this.GetFullTableName($"{this._configuration.EmbeddingsTableName}_{collectionName}")};";
+
+            command.Parameters.AddWithValue("@collectionName", collectionName);
+
+            await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> DoesCollectionExistsAsync(string collectionName,
+        CancellationToken cancellationToken = default)
+    {
+        collectionName = NormalizeIndexName(collectionName);
+
+        var collections = this.GetCollectionsAsync(cancellationToken)
```

Comment:
This could be optimized by using a select that does `$"table_name = {this._configuration.MemoryTableName}_{collectionName}"` for if there is a large number of collections.
Can be in a separate PR though.